chdecompressor/decompressor.go

```diff
@@ -83,7 +84,8 @@ func (r *Reader) readNextBlock() error {
 			return fmt.Errorf("cannot decompress lz4 block: %w", err)
 		}
 	case zstdType:
-		r.data, err = zstd.Decompress(r.data, block)
+		r.data = r.data[:0] // Wipe the slice but keep allocated memory
```

Comment:
Why is this needed? We allocate the slice above.

Reply:
Its because the [new library is doing](https://pkg.go.dev/github.com/klauspost/compress/zstd#Decoder.DecodeAll) : 
> Output will be appended to dst

Compared to datadog which just replace the buffer